src/main/kotlin/org/neo4j/graphql/Translator.kt

```diff
@@ -359,75 +359,79 @@ class Translator(val schema: GraphQLSchema) {
-                .type("Query")
-                { it.dataFetcher("hello") { env -> "Hello ${env.getArgument<Any>("what")}!" } }
+                .type("Query") { it.dataFetcher("hello") { env -> "Hello ${env.getArgument<Any>("what")}!" } }
                 .build()
 
         val schemaGenerator = SchemaGenerator()
-        return schemaGenerator.makeExecutableSchema(typeDefinitionRegistry, runtimeWiring)
-                .transform { bc -> bc.additionalDirectives(additionalDirectives()).build() }
+        return schemaGenerator.makeExecutableSchema(augmentedTypeDefinitionRegistry, runtimeWiring)
                 .transform { sc -> sc.build() } // todo add new queries, filters, enums etc.
     }
 
-    private fun additionalDirectives(): Set<GraphQLDirective> {
-        val directives = setOf(
-                GraphQLDirective.newDirective().name("property").description("additional metadata per field")
-                        .argument{ b -> b.name("name").description("name of the graph property").type(Scalars.GraphQLString)}.build(),
-                GraphQLDirective("relation", "relation directive",
-                EnumSet.of(Introspection.DirectiveLocation.FIELD, Introspection.DirectiveLocation.OBJECT),
-                listOf(GraphQLArgument("name", Scalars.GraphQLString),
-                        GraphQLArgument("direction", "relationship direction", Scalars.GraphQLString, "OUT"),
-                        GraphQLArgument("from", "from field name", Scalars.GraphQLString, "from"),
-                        GraphQLArgument("to", "to field name", Scalars.GraphQLString, "to")), false, false, true))
-        return directives
-    }
-
     private fun augmentSchema(typeDefinitionRegistry: TypeDefinitionRegistry, schemaParser: SchemaParser, ctx: Translator.Context): TypeDefinitionRegistry {
+        val directivesSdl = """
+            enum RelationDirection {
+               IN
+               OUT
+            }
+            directive @relation(name:String, direction: RelationDirection = OUT, from: String = "from", to: String = "to") on FIELD_DEFINITION | OBJECT
+            directive @cypher(statement:String) on FIELD_DEFINITION
+            directive @property(name:String) on FIELD_DEFINITION
+        """
+        typeDefinitionRegistry.merge(schemaParser.parse(directivesSdl))
 
         val augmentations = typeDefinitionRegistry.types().values.filterIsInstance<ObjectTypeDefinition>().map { augmentedSchema(ctx, it) }
 
         val augmentedTypesSdl = augmentations.flatMap { listOf(it.filterType, it.ordering, it.inputType).filter { it.isNotBlank() } }.joinToString("\n")
         typeDefinitionRegistry.merge(schemaParser.parse(augmentedTypesSdl))
 
-        val schemaDefinition = typeDefinitionRegistry.schemaDefinition().orElseGet { SchemaDefinition() }
-        if (!typeDefinitionRegistry.schemaDefinition().isPresent) typeDefinitionRegistry.add(schemaDefinition).ifPresent { throw RuntimeException(it.toSpecification().toString()) }
```

Comment:
it's not fully equivalent to the before, as before we did add the schema if it was not there?